// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.kv_store";
option java_outer_classname = "KV_StoreProto";
option objc_class_prefix = "KVSG";

package kv_store;

// Interface exported by the server.
service KV_Store {
  rpc Put(PutRequest) returns (PutResponse) {}
  rpc Get(GetRequest) returns (GetResponse) {}
  rpc Del(DelRequest) returns (DelResponse) {}
}

message PutRequest {
  string key = 1;
  string data = 2;
}

message PutResponse {
  int32 resCode = 1;
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  int32 resCode = 1;
  string data = 2;
}

message DelRequest {
  string key = 1;
}

message DelResponse {
  int32 resCode = 1;
}

message MyStruct {
  int32  aNumber = 1;
  string aString = 2;
  bytes someBytes = 3;
}